### PR TITLE
COMP: Simplify directory listing using vtksys::Directory

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
Similar to https://github.com/NIRALUser/RigidAlignment/pull/9

File: Modules/CLI/Groups/Groups.cxx [hunk 1]
```diff
@@ -10,41 +10,44 @@
...
-        closedir(dir);
-        sort(list.begin(), list.begin() + list.size());
-        return true;
-    }else{
-        
-        cerr<<"The directory does not exist! "<<path<<endl;
-        return false;
+      continue;
+    }
+    if(filename.find(suffix) != string::npos && filename.find_last_of(suffix) == filename.size() - 1)
```
Comment (Connor-Bowley) [inline Modules/CLI/Groups/Groups.cxx:44]: I don't think the logic here is actually correct (the original logic, which your change replicates).

`std::string::find_last_of` has the following semantics (from https://en.cppreference.com/w/cpp/string/basic_string/find_last_of)

Finds the last character equal to one of characters in the given character sequence.

The current use of this is `if(! getListFile(landmark.c_str(), landmarkList, "vtk"))` which makes it seem like we are looking for a `.vtk` file, but in reality a `.mov` file would incorrectly match.

I think this needs to change to use something like `std::string::rfind` instead.

File: Modules/CLI/Groups/Groups.cxx [hunk 2]
```diff
@@ -10,41 +10,44 @@
...
+    {
+      list.push_back(filepath);
     }
+  }
+  sort(list.begin(), list.begin() + list.size());
```
Comment (Connor-Bowley) [inline Modules/CLI/Groups/Groups.cxx:49]: If we are changing things anyway can we switch this to be `sort(list.begin(), list.end())` to be more idiomatic and clear we are sorting the whole list.

**Review by Connor-Bowley — CHANGES_REQUESTED** [review]

**Comment by Connor-Bowley** [comment]
@jcfr Any idea when you will have time for this, or I should I make the change?

File: Modules/CLI/Groups/Groups.cxx (hunk 1 above)
Comment (Connor-Bowley) [reply Modules/CLI/Groups/Groups.cxx]: I have made this update.

File: Modules/CLI/Groups/Groups.cxx (hunk 2 above)
Comment (Connor-Bowley) [reply Modules/CLI/Groups/Groups.cxx]: Updated.

**Comment by jcfr** [comment]
Thanks for the help 🙏

**Review by Connor-Bowley — APPROVED** [review]

**Pull request merged**